pageserver/src/tenant/storage_layer.rs

```diff
@@ -255,24 +300,44 @@ impl LayerAccessStats {
         ret
     }
 
-    fn most_recent_access_or_residence_event(
-        &self,
-    ) -> Either<LayerAccessStatFullDetails, LayerResidenceEvent> {
+    /// Get the latest access timestamp, falling back to latest residence event, falling back to `fallback`.
+    ///
+    /// The `fallback` must be supplied for the case where there has not yet been a call to the
+    /// [`record_residence_event`] method.
+    /// If `fallback` needs to be used, we log a rate-limited warning in global scope.
+    pub(crate) fn latest_activity<F>(&self, fallback: F) -> SystemTime
```

Comment:
```suggestion
    pub(crate) fn latest_activity<F>(&self) -> Option<SystemTime>
```

Would rather see this instead of new `Option::unwrap_or_else` alike APIs unless we use the value created by the factory for something within this function?